### DISCUSSION TITLE
2 openobserve instances using same s3 (1 reader, 1 writer)

### DISCUSSION THREAD

**Discussion by a3kov** [openobserve/openobserve#4922] [Q&A]
I have this idea. Since in OpenObserve model the server is stateless, I want to get advantage of it. I am planning to have 2 OpenObserve instances running with the same s3 storage:
- writer instance, it will run on a weak VM in the cloud and collect/write logs to S3
- query/browser instance, that will be used to access/query the collected data. I will run it on my desktop machine which is quite beefy compared to the VMs I have.

I realize that I can't access the data not pushed yet to S3 but that is ok. I think I could also close UI access on the cloud instance completely after setup as it will be running as "backend" only.

Assuming my desktop machine internet connection is fast enough, will there be any issues with this setup ? As far as I understand these 2 instances don't even need to know about each other ? Or any synchronization is needed ? I guess I may need a cluster (and shared metadata) if I want to configure functions that work on ingestion while working on the local instance

**Comment by hengfeiyang** [comment]
We have three major component:
- Ingester
- Querier
- Compactor

The `Ingester` is used to collect data and write to s3, the `Compactor` is used to merge small files and delete old data(data retention). The `querier` used to query data.

In your architecture, maybe there are a few issues:
- How to sync the data (metadata) between cloud instance and your local instance, a cloud database service?
- Which instance will be used to run the `compactor` role?

If you run `All-In-One` mode for both instances, the `compactor` will have two instance and maybe cause file duplicated issue.

My suggestion:
1. You can run the cloud instance as a `all-in-one` single instance with a RDS & S3 to maintain the files.
2. The local node only run a `querier` node with `ZO_NODE_ROLE=querier` to fetch data from same s3, but it need connect to the same RDS to get the metadata(stream schema, s3 file list index).

**Reply by a3kov** [reply]
Yeah I was planning to run the compactor on the cloud instance. The expected volume of data is not big, it's just I prefer my queries to always be snappy. I will try to implement this following your suggestion later.
Thanks!
